src/record/record.js

```diff
@@ -151,17 +151,18 @@ Record.prototype.set = function( pathOrData, dataOrCallback, callback ) {
 	var newValue = jsonPath.set( oldValue, path, data, this._options.recordDeepCopy );
 
 	if ( oldValue === newValue ) {
+		if ( typeof callback === 'function' ) utils.setImmediate( callback, null );
```

Comment:
Yes, I would rather see that. Since as it is now, the user of the interface will already be notified with an error that the connection is closed and then it tries to set the data anyway. If offline should be supported, it would be a warning at most stating that you are in offline mode.

So I would recommend moving the connectionState check further up in the set function and do a setImmediate on the callback with the error and then return this.

Is this what you meant?